src/engine/controllers/path_controller_sub_system.cpp

```diff
@@ -0,0 +1,145 @@
+#include "platform/i_platform.h"
+#include "engine/engine.h"
+#include "engine/controllers/path_controller_sub_system.h"
+#include "core/path_controller_component.h"
+#include "core/i_position_component.h"
+#include "core/i_health_component.h"
+#include "core/i_move_component.h"
+#include "core/i_collision_component.h"
+
+namespace engine {
+
+PathControllerSubSystem::PathControllerSubSystem()
+    : mScene(Scene::Get())
+    , mProgramState(core::ProgramState::Get())
+{
+}
+
+
+void PathControllerSubSystem::Init()
+{
+    mPathSystem = engine::Engine::Get().GetSystem<engine::path::PathSystem>();
+}
+
+
+void PathControllerSubSystem::Update(Actor& actor, double DeltaTime)
+{
+    Opt<PathControllerComponent> pathCC=actor.Get<IControllerComponent>();
+    if (!pathCC.IsValid()||!pathCC->IsEnabled())
+    {
+        return;
+    }
+    if (mProgramState.mMode == core::ProgramState::Client)
+    {
+        return;
+    }
+    auto targetHolderC = actor.Get<ITargetHolderComponent>();
+    if (!targetHolderC.IsValid())
+    {
+        return;
+    }
+    auto healthC = actor.Get<IHealthComponent>();
+    if (!healthC.IsValid() || !healthC->IsAlive())
+    {
+        return;
+    }
+    UpdateTarget( actor, targetHolderC );
+    Opt<Actor> currentTarget( mScene.GetActor( targetHolderC->GetTargetGUID() ) );
+    auto moveC( actor.Get<IMoveComponent>() );
+    auto positionC( actor.Get<IPositionComponent>() );
+    if (currentTarget.IsValid())
+    {
+        if (mPathSystem->GetBoxIndex( actor ) == mPathSystem->GetBoxIndex( *currentTarget ))
+        {
+            auto targetPositionC( currentTarget->Get<IPositionComponent>() );
+            glm::vec2 const distV( (targetPositionC->GetX() - positionC->GetX()), (targetPositionC->GetY() - positionC->GetY()) );
+            moveC->SetHeading( atan2( distV.y, distV.x ) );
+        }
+        else
+        {
+            moveC->SetHeading( mPathSystem->GetDirection( actor, *currentTarget ) );
+        }
+        positionC->SetOrientation( moveC->GetHeading() );
+        moveC->SetMoving( true );
+        Opt<IPositionComponent> const targetPositionC = currentTarget->Get<IPositionComponent>();
+        glm::vec2 const Diff( targetPositionC->GetX() - positionC->GetX(), targetPositionC->GetY() - positionC->GetY() );
```

Comment:
is this Diff different from distV? is any of the positions updated in the meantime? if no, then it's probably enough to calc it once (though it seems quite cheap)